tests/hipscat/pixel_tree/test_pixel_tree_builder.py

```diff
@@ -107,6 +201,70 @@ def test_pixel_builder_retrieve_none_node():
     assert tree_builder[HealpixPixel(10, 10)] is None
 
 
+def test_split_leaf_to_match_partitioning():
+    builder1 = PixelTreeBuilder()
+    builder2 = PixelTreeBuilder()
+    builder1.create_node((0, 0), PixelNodeType.LEAF)
+    builder2.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((1, 0), PixelNodeType.INNER)
+    builder2.create_node((2, 0), PixelNodeType.LEAF)
+    builder1.split_leaf_to_match_partitioning(builder2[(0, 0)])
+    assert builder1[(0, 0)].node_type == PixelNodeType.INNER
+    for pixel in [(1, 1), (1, 2), (1, 3)]:
+        assert pixel in builder1
+        assert builder1[pixel].node_type == PixelNodeType.LEAF
+    for pixel in [(2, 0), (2, 1), (2, 2), (2, 3)]:
+        assert pixel in builder1
+        assert builder1[pixel].node_type == PixelNodeType.LEAF
+
+
+def test_split_leaf_not_exist_errors():
+    builder1 = PixelTreeBuilder()
+    builder2 = PixelTreeBuilder()
+    builder2.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((1, 0), PixelNodeType.INNER)
+    builder2.create_node((2, 0), PixelNodeType.LEAF)
+    assert (0, 0) not in builder1
+    with pytest.raises(ValueError, match="No node in tree"):
+        builder1.split_leaf_to_match_partitioning(builder2[(0, 0)])
+
+
+def test_split_leaf_not_leaf_errors():
+    builder1 = PixelTreeBuilder()
+    builder2 = PixelTreeBuilder()
+    builder1.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((1, 0), PixelNodeType.INNER)
+    builder2.create_node((2, 0), PixelNodeType.LEAF)
+    assert builder1[(0, 0)].node_type == PixelNodeType.INNER
+    with pytest.raises(ValueError, match="is not a leaf node"):
+        builder1.split_leaf_to_match_partitioning(builder2[(0, 0)])
+
+
+def test_add_all_descendants_from_node():
+    builder1 = PixelTreeBuilder()
+    builder2 = PixelTreeBuilder()
+    builder1.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((0, 0), PixelNodeType.INNER)
+    builder2.create_node((1, 0), PixelNodeType.INNER)
```

Comment:
It would be worth also adding a leaf node (1, 1) here to check multiple nodes at the same level.